### ISSUE TITLE
Overload completion called on background thread

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#6021]
VS Mac 7.7.0.1351
Git revision: 419d9b2d9f81609106d5aac4257437c1dc79d553
Build date: 2018-09-18 07:43:59+00
Build branch: release-7.7
Xamarin extensions: 017d5b0ae3e151e427b8e25a67eb88e9f319d40e

Operation run on background thread triggering assert. Not sure how to repro.

```
Error while adding overload : [CompletionData: Icon=md-newmethod, DisplayText=HandleAction, Description=, CompletionText=HandleAction, DisplayFlags=None]
System.InvalidOperationException: Operation not supported in background thread
  at MonoDevelop.Core.Runtime.AssertMainThread () [0x00012] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:450 
  at MonoDevelop.Ide.Editor.TextEditor.get_IndentationTracker () [0x00000] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs:1545 
  at MonoDevelop.CSharp.Completion.Provider.DelegateCompletionProvider+<GetInsertText>d__23.MoveNext () [0x00069] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/DelegateCompletionProvider.cs:417 
--- End of stack trace from previous location where exception was thrown ---
  at MonoDevelop.CSharp.Completion.Provider.DelegateCompletionProvider+<GetTextChangeAsync>d__22.MoveNext () [0x00029] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/DelegateCompletionProvider.cs:408 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.Completion.CommonCompletionProvider+<TryAddSnippetInvocationPart>d__3.MoveNext () [0x0006c] in /_/src/Features/Core/Portable/Completion/CommonCompletionProvider.cs:55 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.Completion.CommonCompletionProvider+<GetDescriptionAsync>d__2.MoveNext () [0x000d0] in /_/src/Features/Core/Portable/Completion/CommonCompletionProvider.cs:42 
--- End of stack trace from previous location where exception was thrown ---
  at MonoDevelop.Ide.CodeCompletion.RoslynCompletionData+<CreateTooltipInformation>d__39.MoveNext () [0x000b1] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/RoslynCompletionData.cs:312 
--- End of stack trace from previous location where exception was thrown ---
  at MonoDevelop.Ide.CodeCompletion.TooltipInformationWindow+<AddOverload>d__11.MoveNext () [0x00036] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.CodeCompletion/TooltipInformationWindow.cs:87 
```

> VS bug [#686500](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/686500)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.524 (master)

Pull Request #6027 merged by: Lluis Sanchez
Author: mono
Commit: cfcb4b192aac1512b3055a26ace3200697e8ccc5 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1361 (release-7.7)

Pull Request #6028 merged by: Lluis Sanchez
Author: mono
Commit: 7922720e6d668851f551500c5830ad650feb0dc4 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #6027 merged by: Lluis Sanchez
Author: mono
Commit: cfcb4b192aac1512b3055a26ace3200697e8ccc5 (xamarin/monodevelop)
